src/appshell/internal/appshellconfiguration.cpp

```diff
@@ -78,19 +85,25 @@ void AppShellConfiguration::setHasCompletedFirstLaunchSetup(bool has)
     settings()->setSharedValue(HAS_COMPLETED_FIRST_LAUNCH_SETUP, Val(has));
 }
 
-StartupModeType AppShellConfiguration::startupModeType() const
+ValCh<StartupModeType> AppShellConfiguration::startupModeType() const
 {
-    return settings()->value(STARTUP_MODE_TYPE).toEnum<StartupModeType>();
+    ValCh<StartupModeType> result;
+    result.ch = m_startupModeTypeChanged;
+    result.val = settings()->value(STARTUP_MODE_TYPE).toEnum<StartupModeType>();
+    return result;
 }
```

Comment:
It might be more convenient to make the `ValCh` a member field. Then this method can just return the field.
In the `init` method, you'd do
```c++
    m_startupModeType.val = settings()->value(STARTUP_MODE_TYPE).toEnum<StartupModeType>();
    settings()->valueChanged(STARTUP_MODE_TYPE).onReceive(nullptr, [this](const Val& val) {
        m_startupModeType.set(val.toEnum<StartupModeType>());
    });
```
Note the `.set` method of ValCh.